### PR TITLE
feat: add CODER_OIDC_IGNORE_EMAIL_VERIFIED config knob

### LINKED ISSUE #5065: bug: allow ignoring OIDC email_verified claim
# Problem

(Context: see https://github.com/coder/coder/issues/3954)

Coder will check for the `email_verified` OIDC claim if it is provided, and block sign-in if the user's email is not verified.
Users with unverified emails atttempting to login will get a message similar to 

```
{"message":"Verify the \"user@acme.corp\" email address on your OIDC provider to authenticate!"}
```

From the Auth0 docs:

https://auth0.com/docs/manage-users/user-accounts/user-profiles/verified-email-usage

> In case of federated identity providers, they sometimes report if the user has a verified email, and based on that, Auth0 sets the email_verified field in the user profile. This, however, transfers the responsibility to the identity provider to do it properly - something we can't ensure. We also don't know if the verified email from that provider is still owned by the user.

> When users authenticate with a federated identity provider (e.g. a social or enterprise connection), the value of the email_verified field will match what the identity provider returns in the user profile. If they identity provider does not return any value, it will be set to false.

This means that certain IdPs through Auth0 (for example, AzureAD) are essentially unusable with Coder currently -- if the IdP does not return the `email_verified` claim, Auth0 will default it to `false` and Coder will block sign-in. 

# Solution

Add a command-line flag / environment variable `CODER_OIDC_IGNORE_EMAIL_VERIFIED`.
This will default to false. If it is set to true, skip the code path to check the `email_verified` claim:

https://github.com/coder/coder/blob/4b3d211e004ec5e4a3b30f98ca3c3e8ec687168a/coderd/userauth.go#L251-260

This will essentially assume that all emails are verified.

### PR DESCRIPTION
**Pull request by johnstcn**
Fixes #5065

### CODE REVIEW

**Comment by johnstcn** [comment]
I have read the CLA Document and I hereby sign the CLA

File: cli/deployment/config_test.go [hunk 1]
```diff
@@ -125,20 +125,22 @@ func TestConfig(t *testing.T) {
...
+			"CODER_OIDC_SCOPES":                "something,here",
+			"CODER_OIDC_IGNORE_EMAIL_VERIFIED": "true",
 		},
 		Valid: func(config *codersdk.DeploymentConfig) {
 			require.Equal(t, config.OIDC.IssuerURL.Value, "https://accounts.google.com")
 			require.Equal(t, config.OIDC.EmailDomain.Value, "coder.com")
 			require.Equal(t, config.OIDC.ClientID.Value, "client")
 			require.Equal(t, config.OIDC.ClientSecret.Value, "secret")
-			require.Equal(t, config.OIDC.AllowSignups.Value, false)
+			require.False(t, config.OIDC.AllowSignups.Value, false)
```
Comment (mtojek) [inline cli/deployment/config_test.go:141]: nit: `require.False(t, config.OIDC.AllowSignups.Value)`

File: cli/deployment/config_test.go [hunk 2]
```diff
@@ -125,20 +125,22 @@ func TestConfig(t *testing.T) {
...
 			require.Equal(t, config.OIDC.Scopes.Value, []string{"something", "here"})
+			require.Equal(t, config.OIDC.IgnoreEmailVerified.Value, true)
```
Comment (mtojek) [inline cli/deployment/config_test.go:143]: nit: `require.True`

**Comment by johnstcn** [comment]
recheck

File: cli/deployment/config_test.go (hunk 1 above)
Comment (johnstcn) [inline cli/deployment/config_test.go:141]:
```suggestion
			require.Equal(t, config.OIDC.AllowSignups.Value, false)
```

File: coderd/userauth.go [hunk 3]
```diff
@@ -261,7 +264,7 @@ func (api *API) userOIDC(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 	verifiedRaw, ok := claims["email_verified"]
-	if ok {
+	if ok && !api.OIDCConfig.IgnoreEmailVerified {
```
Comment (mtojek) [inline coderd/userauth.go:267]: nit: maybe add a log entry here to inform that verification failed, but the code will ignore it.

**Review by mtojek — APPROVED** [review]
LGTM, left some nit-picks

File: cli/deployment/config_test.go (hunk 1 above)
Comment (mafredri) [inline cli/deployment/config_test.go:141]: This still has the `, false` although it changed from Equal to False, intentional?

File: docs/admin/auth.md [hunk 4]
```diff
@@ -76,11 +76,21 @@ Once complete, run `sudo service coder restart` to reboot Coder.
...
 
 If your OpenID Connect provider requires client TLS certificates for authentication, you can configure them like so:
+
 ```console
 CODER_TLS_CLIENT_CERT_FILE=/path/to/cert.pem
 CODER_TLS_CLIENT_KEY_FILE=/path/to/key.pem
 ```
 
+Coder requires all OIDC email addresses to be verified by default. If the `email_verified` claim is present in the token response from the identity provider, Coder will validate that its value is `true`.
+If you need to modify this behaviour, you can disable this behaviour like so:
```
Comment (mafredri) [inline docs/admin/auth.md:86]: This part felt a bit repetitive, spiffed it up a bit and US-Englishified it 😅

```suggestion
If needed, you can disable this behavior with the following setting:
```

**Review by mafredri — APPROVED** [review]
Looks good! 👍

File: cli/deployment/config_test.go (hunk 1 above)
Comment (johnstcn) [reply cli/deployment/config_test.go:141]: Nope, although `require.False` is more idiomatic regardless. Updated.

**Pull request merged**